waiter/src/waiter/process_request.clj

```diff
@@ -590,17 +619,37 @@
                               (deliver reservation-status-promise :generic-error)
                               ; close request-state-chan to mark the request as finished
                               (async/close! request-state-chan)
-                              ; Handle error lower down
-                              (throw e))))))
+                              (let [response (-> (process-exception-fn track-process-error-metrics request descriptor e)
+                                                 (update :headers (fn [headers]
+                                                                    (merge @response-headers headers))))
+                                    request (-> request
+                                                (utils/mark-request-time :closed))]
+                                (rlog/log (assoc (rlog/request->context request)
+                                                 :status (:status response)
+                                                 :termination-state :backend-error))
+                                response))))))
                     (catch Exception e
-                      ; Handle error lower down
-                      (throw (ex-info "Rethrowing exception from process" {:descriptor descriptor :source :pr/process} e)))))))
+                      (let [{:keys [descriptor]} (ex-data e)
+                            response (-> (process-exception-fn track-process-error-metrics request descriptor e)
+                                         (update :headers (fn [headers]
+                                                            (merge @response-headers headers))))
```

Comment:
What's the effect of using `process-exception-fn` instead of `throw`ing?

Reply:
Lots has changed here.  Not sure if the question still applies.